Rework steps + links input in edit mode

MarcelBolten opened the pull request:
* move input to top: the lists can get long and currently one needs to scroll a lot
* make lists toggelable, this requires expansion of js toggle-next action

old:
![image](https://user-images.githubusercontent.com/65481677/177863033-d8df8fe9-4123-4f83-bb48-4aef6e095a17.png)
new:
![image](https://user-images.githubusercontent.com/65481677/177862681-03e3eb07-02c1-47fa-885d-2d2c610e0804.png)
![image](https://user-images.githubusercontent.com/65481677/177862797-a930e249-51c8-4bd3-9bd2-f37d767d39eb.png)

old:
![image](https://user-images.githubusercontent.com/65481677/177863087-fee493cc-f9d8-4662-b455-d447dc95e87e.png)
new:
![image](https://user-images.githubusercontent.com/65481677/177862723-9c9d4293-4e77-45dd-8ea1-e1007bafbeb2.png)
![image](https://user-images.githubusercontent.com/65481677/177862769-3fb5eb37-2b03-4dbd-a7db-040c7b6ca4d7.png)

NicolasCARPi commented:
While you're in that part of the code, it would be great if the link input would use the same kind of select than say the team select on register page.

MarcelBolten commented:
Are you talking about the category select element, the actual input that is using  autocomplete or both that should use the selectpicker from bootstrap-select?
Probably the category select, right?

NicolasCARPi commented:
No the actual "title" input. There should not be thousands of categories, but there might very well be thousands of items, so the idea is to provide an easier way to find what you want to link to without necessarily typing/knowing the full title.

MarcelBolten commented:
I will open a new PR in a few days to address your request but would like to keep it separated from this one.

MarcelBolten commented:
> there might very well be thousands of items, so the idea is to provide an easier way to find what you want to link to without necessarily typing/knowing the full title.

Should this still be dynamically loaded  via ajax based on user input but without limiting the number of entries or do you want the full database (id, category, title, maybe the date too if two titles are the same?) (if no category is selected) to be loaded upfront?

NicolasCARPi commented:
> Should this still be dynamically loaded via ajax based on user input but without limiting the number of entries or do you want the full database (id, category, title, maybe the date too if two titles are the same?) (if no category is selected) to be loaded upfront?

Yes dynamically loaded, but maybe some way to list entries without having to search? The problem is that currently one needs to start typing something and it would be nice to be able to simply browse the current entries (maybe ordered by last time modified?).

Also note that I have write access to the bootstrap-select repository so if there is a change that needs merging, let me know.

NicolasCARPi reviewed (changes requested):
I think the toggle should encompass the input + the steps, seems a bit weird that the input stays visible. For instance, it is confusing to add something, but it doesn't show up because the "content" is not shown, only input.

Also, it seems the localStorage to keep state around is not working properly.

MarcelBolten commented:
> I think the toggle should encompass the input + the steps, seems a bit weird that the input stays visible. For instance, it is confusing to add something, but it doesn't show up because the "content" is not shown, only input.

My reasoning is that if this is a long list the user will not see the newly added item anyways because the entry might be added to the very bottom. I added a notification if the lists of steps and/or links are hidden but you have the last word.

> Also, it seems the localStorage to keep state around is not working properly.

Fixed.

NicolasCARPi commented on src/ts/langs/en_GB.ts:
```diff
@@ -24,6 +24,10 @@ const t = {
     "tag-delete-warning": "Delete this tag?",
     "template-title": "Template title",
     "upload-file-comment": "File comment",
+    "step-added": "Step added",
```
I'm wary of adding new terms to translate. I think here having "Saved" or nothing is enough. 4 terms times 16 languages is 64 strings that need work to translate.

NicolasCARPi reviewed (approved):
:+1:
